### PR TITLE
Replace busy loops (pow submission & node) to conditional variable

### PR DESCRIPTION
**Pull request by ckyang**
https://github.com/Zilliqa/Zilliqa/issues/252
https://github.com/Zilliqa/Issues/issues/47

### CODE REVIEW

**Comment by ckyang** [comment]
@Gnnng Removed.

**Comment by ckyang** [comment]
Un-related code removed.
Still need to use this PR for implementing https://github.com/Zilliqa/Issues/issues/47

**Comment by bb111189** [comment]
Ok. Once ready for review, remove the wip message from this pr.

File: src/libDirectoryService/PoW2Processing.cpp [hunk 1]
```diff
@@ -197,37 +197,25 @@ bool DirectoryService::ProcessPoW2Submission(
...
-            this_thread::sleep_for(
-                chrono::milliseconds(sleep_time_while_waiting));
+            LOG_EPOCH(WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+                      "Time out while waiting for state transition and "
+                      "consensus object creation ");
         }
+
+        LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                  "State transition is completed and consensus object "
+                  "creation. (check for timeout)");
```
Comment (Gnnng) [inline src/libDirectoryService/PoW2Processing.cpp]: Can this log message be rephrased a bit? It doesn't read well now. Others similar ones also need some change.

**Pull request merged**